### PR TITLE
[nrf noup] Added support for SPI NOR external flash

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Added configs allowing to enable external flash and DFU using SPI NOR for application and mcuboot images.

### CODE REVIEW

File: config/nrfconnect/chip-module/Kconfig.mcuboot.defaults [hunk 1]
```diff
@@ -52,6 +52,34 @@ config FPROTECT
...
+    default 4096
+
+config MULTITHREADING
+    bool
+    default y
+
+endif
+
+# All boards beside nRF7002DK use QSPI NOR external flash
+if BOARD_NRF5340DK_NRF5340_CPUAPP || BOARD_NRF52840DK_NRF52840
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.mcuboot.defaults:81]: What about the dongle or customers' boards? Maybe better to simply use `!BOARD_NRF7002DK_NRF5340_CPUAPP`

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 2]
```diff
@@ -201,10 +201,25 @@ config CHIP_OTA_REQUESTOR
     bool
     default y
 
+# All boards beside nRF7002DK use QSPI NOR external flash
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.defaults:204]:
```suggestion
# All boards besides nRF7002DK use QSPI NOR external flash
```

File: config/nrfconnect/chip-module/Kconfig.features [hunk 3]
```diff
@@ -58,9 +58,36 @@ config NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE
...
 
+config CHIP_SPI_NOR
+	bool "Enable SPI NOR feature set"
+	help
+		Enables SPI NOR with the set of options
+		configuring pages and buffer sizes.
+
+if CHIP_SPI_NOR
+
+config SPI
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.features:69]: Use `imply` in `CHIP_SPI_NOR`?

File: config/nrfconnect/chip-module/Kconfig.mcuboot.defaults (hunk 1 above)
Comment (kkasperczyk-no) [reply config/nrfconnect/chip-module/Kconfig.mcuboot.defaults:81]: I wasn't sure what is the best option. On the other hand dongle and customer boards may not support QSPI either.

Comment (Damian-Nordic) [reply config/nrfconnect/chip-module/Kconfig.mcuboot.defaults:81]: You're right. Better not to set anything if we don't know what is supposed to be used.

Comment (kkasperczyk-no) [reply config/nrfconnect/chip-module/Kconfig.mcuboot.defaults:81]: I changed it also in other places, as previously I wasn't consistent.

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**